### PR TITLE
Fix incorrect singularization for words ending with -ves (arrives/drives/curves/behaves) and add tests

### PR DESCRIPTION
**Pull request by jesi318**
Here is a checklist you should tick through before submitting a pull request: 
 - [x] Implementation is clean
 - [x] Code adheres to the existing coding standards; e.g. no curlies for one-line blocks, no redundant empty lines between methods or code blocks, spaces rather than tabs, etc.
 - [x] No Code Analysis warnings
 - [x] There is proper unit test coverage
 - [x] If the code is copied from StackOverflow (or a blog or OSS) full disclosure is included. That includes required license files and/or file headers explaining where the code came from with proper attribution
 - [x] There are very few or no comments (because comments shouldn't be needed if you write clean code)
 - [x] Xml documentation is added/updated for the addition/change
 - [x] Your PR is (re)based on top of the latest commits from the `main` branch (more info below)
 - [x] Link to the issue(s) you're fixing from your PR description. Use `fixes #<the issue number>`
 - [x] Readme is updated if you change an existing feature or add a new one
 - [x] Run either `build.cmd` or `build.ps1` and ensure there are no test failures
 
 
Summary

Fixes a bug where arrives.Singularize() → arrife (and similar: drives → drife, curves → curf, behaves → behafe etc) by adjusting the singularization rule set and ordering.
Adds unit tests that assert correct singular forms for common -ves cases and true -fe exceptions.

- **fixes** #1674 

Why this change was made/ Root Cause
_default.AddSingular("([^f])ves$", "$1fe"); is a greedy rule which matched other strings mentioned in the issue above

What I changed

Reworked -ves singular rules in Vocabularies.Defaultso exceptions for nouns that map to -f/-fe (knife→knives, wife→wives, leaf→leaves, etc.) are handled before the general catch-all that converts -ves → -ve (verbs/adjectives like arrive/drive/curve/behave).
Replaced overly-generic pattern(s) that caused verbs to be mis-singularized with a combination of:

-  Explicit stem-based rules for known -f/-fe nouns, and

- A general (.+)ves$ -> $1ve catch-all for remaining -ves cases.

- Corrected the incorrect test  yield return ["safe", "saves"]; to yield return ["safe", "safes"]; (which makes it grammattically correct according to me) and added a rule for it since it was an edge case . Please let me know if I'm wrong.

-  Added a focused unit test method Singularize_Ves_EdgeCases in InflectorTests.cs

-  ([lr])ves$ → $1f was separated because this keeps this broad heuristic distinct from explicit stem-based rules, preventing it from interfering with more precise matches and making rule behavior easier to control and debug.

### CODE REVIEW

File: src/Humanizer/Inflections/Vocabularies.cs [hunk 1]
```diff
@@ -43,10 +43,11 @@ static Vocabulary BuildDefault()
         _default.AddSingular("(n)ews$", "$1ews");
         _default.AddSingular("([dti])a$", "$1um");
         _default.AddSingular("(analy|ba|diagno|parenthe|progno|synop|the|ellip|empha|neuro|oa|paraly)ses$", "$1sis");
-        _default.AddSingular("([^f])ves$", "$1fe");
+        _default.AddSingular("(.+)ves$", "$1ve");
         _default.AddSingular("(hive)s$", "$1");
         _default.AddSingular("(tive)s$", "$1");
-        _default.AddSingular("([lr]|hoo|lea|loa|thie)ves$", "$1f");
+        _default.AddSingular("(lea|loa|cal|hal|sel|el|shel|wol|thie|dwar|hoo)ves$", "$1f");
```
[the remark this answers is not in the document]
Comment (jesi318) [reply src/Humanizer/Inflections/Vocabularies.cs]: fixed by adding the [lr] rule back and added test for the same

File: src/Humanizer/Inflections/Vocabularies.cs [hunk 2]
```diff
@@ -43,10 +43,12 @@ static Vocabulary BuildDefault()
         _default.AddSingular("(n)ews$", "$1ews");
         _default.AddSingular("([dti])a$", "$1um");
         _default.AddSingular("(analy|ba|diagno|parenthe|progno|synop|the|ellip|empha|neuro|oa|paraly)ses$", "$1sis");
-        _default.AddSingular("([^f])ves$", "$1fe");
+        _default.AddSingular("(.+)ves$", "$1ve");
         _default.AddSingular("(hive)s$", "$1");
         _default.AddSingular("(tive)s$", "$1");
-        _default.AddSingular("([lr]|hoo|lea|loa|thie)ves$", "$1f");
+        _default.AddSingular("([lr])ves$", "$1f");
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/Inflections/Vocabularies.cs]: > The new `([lr])ves$ -> $1f` rule still over-matches common `*rves/*lves` words.

Addressed in 95dde187: singularization now uses a finite list of true `-f/-fe → -ves` stems, while `serves`, `solves`, and `curves` fall through to the regular `-ve` result.

File: src/Humanizer/Inflections/Vocabularies.cs [hunk 3]
```diff
@@ -92,7 +94,9 @@ static Vocabulary BuildDefault()
         _default.AddIrregular("tie", "ties", matchEnding: false);
         _default.AddIrregular("lens", "lenses");
         _default.AddIrregular("clove", "cloves");
+        _default.AddIrregular("curve", "curves");
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/Inflections/Vocabularies.cs]: > The new `([lr])ves$ -> $1f` rule still over-matches common `*rves/*lves` words.

Addressed in 95dde187: the broad rule and the `curve/curves` workaround are gone. Explicit exception stems now handle true `-f/-fe` nouns, and the general `-ves → -ve` rule handles `curves`.

File: src/Humanizer/Inflections/Vocabularies.cs [hunk 4]
```diff
@@ -92,7 +94,9 @@ static Vocabulary BuildDefault()
...
         _default.AddIrregular("valve", "valves");
+        _default.AddIrregular("safe", "safes");
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/Inflections/Vocabularies.cs]: > Adding an irregular for “safe/safes” special-cases a regular pluralization.

Addressed at the root in 95dde187: the overly broad `-fe → -ves` plural rule is now an explicit exception list, so `safe/safes` and `cafe/cafes` work through the regular rule and the `safe` irregular has been removed.

File: tests/Humanizer.Tests/InflectorTests.cs [hunk 5]
```diff
@@ -60,6 +60,23 @@ public void SingularizeWordsWithUnknownSingularity(string singular, string plura
...
+    [InlineData("survives", "survive")]
+    [InlineData("derives", "derive")]
+    [InlineData("knives", "knife")]
+    [InlineData("wives", "wife")]
+    [InlineData("leaves", "leaf")]
+    [InlineData("lives", "life")]
+    [InlineData("scarves", "scarf")]
+    [InlineData("hooves", "hoof")]
+    [InlineData("thieves", "thief")]
+    public void Singularize_Ves_EdgeCases(string plural, string expected) =>
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply tests/Humanizer.Tests/InflectorTests.cs]: > The new test method name uses underscore separators.

Addressed in 95dde187: the theory is now named `SingularizeVesEdgeCases`, matching the surrounding test convention.

File: src/Humanizer/Inflections/Vocabularies.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/Inflections/Vocabularies.cs]: > Adding explicit regression tests for representative `*rves/*lves` cases that should resolve to `*ve` would help prevent future rule-order regressions.

Addressed in 95dde187: `serves → serve` and `solves → solve` regressions were added alongside the existing `-f/-fe` noun cases, plus regular `safe`, `cafe`, and `golf` coverage.

File: src/Humanizer/Inflections/Vocabularies.cs [hunk 6]
```diff
@@ -43,10 +44,11 @@ static Vocabulary BuildDefault()
         _default.AddSingular("(n)ews$", "$1ews");
         _default.AddSingular("([dti])a$", "$1um");
         _default.AddSingular("(analy|ba|diagno|parenthe|progno|synop|the|ellip|empha|neuro|oa|paraly)ses$", "$1sis");
-        _default.AddSingular("([^f])ves$", "$1fe");
+        _default.AddSingular("(.+)ves$", "$1ve");
         _default.AddSingular("(hive)s$", "$1");
         _default.AddSingular("(tive)s$", "$1");
-        _default.AddSingular("([lr]|hoo|lea|loa|thie)ves$", "$1f");
+        _default.AddSingular("(cal|dwar|el|hal|hoo|lea|loa|scar|sel|shel|thie|whar|wol)ves$", "$1f");
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/Inflections/Vocabularies.cs]: Addressed in 30ee0941: `tur` is now included symmetrically in the finite `f → ves` and `ves → f` rules, preserving `turf ↔ turves`. Regression coverage was added through the existing inflector class data, and the redundant older `hoof/leaf/loaf/thief` plural rule was removed.

**Comment by clairernovotny** [comment]
Addressed the final `-ves` ambiguity in 5c31bb15: `olive/olives` is now an exact irregular (`matchEnding: false`), so `lives → life` remains supported without turning `olives` into `olife`. Added round-trip regression coverage. All net8/net10/net11 tests, Release packing, and full format verification pass.
